### PR TITLE
[image_gui] Make it a cleaner SOFA module

### PR DESCRIPTION
**Pull request by alxbilger**
- Fixed the module name
- config.h is now config.h.in

A lot more work to do if we want to clean this plugin

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: applications/plugins/image/image_gui/CMakeLists.txt [hunk 1]
```diff
@@ -42,9 +42,10 @@ add_definitions("-DSOFA_BUILD_IMAGE_GUI")
...
 target_link_libraries(${PROJECT_NAME} image ${OPENGL_LIBRARIES} Sofa.GUI.Qt)
 
-sofa_add_targets_to_package(
-    PACKAGE_NAME image
+sofa_create_package_with_targets(
+    PACKAGE_NAME ${PROJECT_NAME}
+    PACKAGE_VERSION ${Sofa_VERSION}
     TARGETS ${PROJECT_NAME} AUTO_SET_TARGET_PROPERTIES
-    INCLUDE_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}"
+    INCLUDE_SOURCE_DIR "."
```
Comment (bakpaul) [inline applications/plugins/image/image_gui/CMakeLists.txt]: I don't understand why this works, it should behave the same as the with the parameter before the modification

Comment (alxbilger) [reply applications/plugins/image/image_gui/CMakeLists.txt]: the macro is not the same

Comment (bakpaul) [reply applications/plugins/image/image_gui/CMakeLists.txt]: Yes but at the end what fixes the include dir is the same in `sofa_auto_set_target_include_directories` : 
```cmake
if(ARG_INCLUDE_SOURCE_DIR)
    if(IS_ABSOLUTE "${ARG_INCLUDE_SOURCE_DIR}")
        set(include_source_root "${ARG_INCLUDE_SOURCE_DIR}")
    else()
        set(include_source_root "${CMAKE_CURRENT_SOURCE_DIR}/${ARG_INCLUDE_SOURCE_DIR}")
    endif()
endif()
```
And regarding your input, the final include is indeed `${CMAKE_CURRENT_SOURCE_DIR}` so it seems that you are not using your include dir when including header with prefix `image_gui`...

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

**Review by fredroy — APPROVED** [review]

**Pull request merged**